lua/lualine/component.lua

```diff
@@ -108,21 +108,32 @@ function M:apply_highlights(default_highlight)
   end
 end
 
----apply icon in front of component (prepemds component with icon)
+---apply icon to component (appends/prepemds component with icon)
 function M:apply_icon()
   local icon = self.options.icon
   if self.options.icons_enabled and icon then
     if type(icon) == 'table' then
+      self.options.icon.align = icon.align
       icon = icon[1]
     end
-    if self.options.icon_color_highlight then
+    if self.options.icon_color_highlight and type(self.options.icon) == 'table' and self.options.icon.align == 'right' then
+      self.status = table.concat {
+        self.status,
+        ' ',
+        self:format_hl(self.options.icon_color_highlight),
+        icon,
+        self:get_default_hl(),
+      }
+    elseif self.options.icon_color_highlight then
       self.status = table.concat {
         self:format_hl(self.options.icon_color_highlight),
         icon,
         self:get_default_hl(),
         ' ',
         self.status,
       }
+    elseif self.options.icon_alignment == 'right' then
```

Comment:
We don't have `options.icon_alignment` any more right ?

Reply:
yup, wasn't on top of it and missed that one.